iceoryx_binding_c/include/iceoryx_binding_c/publisher.h

```diff
@@ -24,20 +24,29 @@
 /// @brief publisher handle
 typedef struct cpp2c_Publisher* iox_pub_t;
 
+/// @brief options to be set for a publisher
+typedef struct
+{
+    // size of the history chunk queue
+    uint64_t historyCapacity;
+
+    // name of the node the publisher belongs to
+    const char* nodeName;
```

Comment:
I also thought about this, but we pay with ease of use and make it potentially more error prone because `nodeName = "foo";`  is not possible anymore and `strncpy` or `strcpy` must be used. So I would opt in for the ease of use and catch too long names internally.

Reply:
I am not in favor of arrays with a fixed length here because it is more cumbersome to use and does not really buy you anything regarding securing the code. Internally it is truncated anyway.